File: scala-play/app/services/EchoService.scala
package services

import javax.inject._
import scala.concurrent._
import models._

import java.time.LocalDate
import java.util.UUID

@Singleton
class EchoService @Inject()()(implicit ec: ExecutionContext) extends IEchoService {
  import IEchoService._

  override def echoBodyString(body: String): Future[EchoBodyStringResponse] = Future {
    EchoBodyStringResponse.Ok(body)
  }

  override def echoBody(body: Message): Future[EchoBodyResponse] = Future {
    EchoBodyResponse.Ok(body)
  }

  override def echoQuery(
                          intQuery: Int,
                          longQuery: Long,
                          floatQuery: Float,
                          doubleQuery: Double,
                          decimalQuery: BigDecimal,
                          boolQuery: Boolean,
                          stringQuery: String,
                          stringOptQuery: Option[String],
                          stringDefaultedQuery: String,
                          stringArrayQuery: List[String],
                          uuidQuery: java.util.UUID,
                          dateQuery: java.time.LocalDate,
                          dateArrayQuery: List[java.time.LocalDate],
                          datetimeQuery: java.time.LocalDateTime,
                          enumQuery: Choice): Future[EchoQueryResponse] = Future {
    EchoQueryResponse.Ok(
      Parameters(
        intField = intQuery,
        longField = longQuery,
        floatField = floatQuery,
        doubleField = doubleQuery,
        decimalField = decimalQuery,
        boolField = boolQuery,
        stringField = stringQuery,
        stringOptField = stringOptQuery,
        stringDefaultedField = stringDefaultedQuery,
        stringArrayField = stringArrayQuery,
        uuidField = uuidQuery,
        dateField = dateQuery,
        dateArrayField = dateArrayQuery,
        datetimeField = datetimeQuery,
        enumField = enumQuery))
  }

  override def echoHeader(
                           intHeader: Int,
                           longHeader: Long,
                           floatHeader: Float,
                           doubleHeader: Double,
                           decimalHeader: BigDecimal,
                           boolHeader: Boolean,
                           stringHeader: String,
                           stringOptHeader: Option[String],
                           stringDefaultedHeader: String,
                           stringArrayHeader: List[String],
                           uuidHeader: java.util.UUID,
                           dateHeader: java.time.LocalDate,
                           dateArrayHeader: List[java.time.LocalDate],
                           datetimeHeader: java.time.LocalDateTime,
                           enumHeader: Choice): Future[EchoHeaderResponse] = Future {
    EchoHeaderResponse.Ok(
      Parameters(
        intField = intHeader,
        longField = longHeader,
        floatField = floatHeader,
        doubleField = doubleHeader,
        decimalField = decimalHeader,
        boolField = boolHeader,
        stringField = stringHeader,
        stringOptField = stringOptHeader,
        stringDefaultedField = stringDefaultedHeader,
        stringArrayField = stringArrayHeader,
        uuidField = uuidHeader,
        dateField = dateHeader,
        dateArrayField = dateArrayHeader,
        datetimeField = datetimeHeader,
        enumField = enumHeader))
  }

  override def echoUrlParams(
                              intUrl: Int,
                              longUrl: Long,
                              floatUrl: Float,
                              doubleUrl: Double,
                              decimalUrl: BigDecimal,
                              boolUrl: Boolean,
                              stringUrl: String,
                              uuidUrl: java.util.UUID,
                              dateUrl: java.time.LocalDate,
                              datetimeUrl: java.time.LocalDateTime,
                              enumUrl: Choice): Future[EchoUrlParamsResponse] = Future {
    EchoUrlParamsResponse.Ok(
      UrlParameters(
        intField = intUrl,
        longField = longUrl,
        floatField = floatUrl,
        doubleField = doubleUrl,
        decimalField = decimalUrl,
        boolField = boolUrl,
        stringField = stringUrl,
        uuidField = uuidUrl,
        dateField = dateUrl,
        datetimeField = datetimeUrl,
        enumField = enumUrl))
  }

  override def echoEverything(
                               uuidHeader: java.util.UUID,
                               datetimeHeader: java.time.LocalDateTime,
                               body: Message,
                               dateUrl: java.time.LocalDate,
                               decimalUrl: BigDecimal,
                               floatQuery: Float,
                               boolQuery: Boolean): Future[EchoEverythingResponse] = Future {
    EchoEverythingResponse.Ok(
      Everything(
        bodyField = body,
        floatQuery = floatQuery,
        boolQuery = boolQuery,
        uuidHeader = uuidHeader,
        datetimeHeader = datetimeHeader,
        dateUrl = dateUrl,
        decimalUrl = decimalUrl,
      ))
  }

  override def sameOperationName(): Future[SameOperationNameResponse] = Future {
    SameOperationNameResponse.Ok()
  }
}

File: ts-express/src/echo_service.ts
import * as service from './spec/echo_service'
import * as models from './spec/models'

export let echoService = (): service.EchoService => {
    let echoBodyString = async (params: service.EchoBodyStringParams): Promise<string> => {
        return params.body
    }

    let echoBody = async (params: service.EchoBodyParams): Promise<models.Message> => {
        return {int_field: params.body.int_field, string_field: params.body.string_field}
    }

    let echoQuery = async (params: service.EchoQueryParams): Promise<models.Parameters> => {
        return {
            int_field: params.int_query,
            long_field: params.long_query,
            float_field: params.float_query,
            double_field: params.double_query,
            decimal_field: params.decimal_query,
            bool_field: params.bool_query,
            string_field: params.string_query,
            string_opt_field: params.string_opt_query,
            string_defaulted_field: params.string_defaulted_query,
            string_array_field: params.string_array_query,
            uuid_field: params.uuid_query,
            date_field: params.date_query,
            date_array_field: params.date_array_query,
            datetime_field: params.datetime_query,
            enum_field: params.enum_query,
        }
    }

    let echoHeader = async (params: service.EchoHeaderParams): Promise<models.Parameters> => {
        return {
            int_field: params['int-header'],
            long_field: params['long-header'],
            float_field: params['float-header'],
            double_field: params['double-header'],
            decimal_field: params['decimal-header'],
            bool_field: params['bool-header'],
            string_field: params['string-header'],
            string_opt_field: params['string-opt-header'],
            string_defaulted_field: params['string-defaulted-header'],
            string_array_field: params['string-array-header'],
            uuid_field: params['uuid-header'],
            date_field: params['date-header'],
            date_array_field: params['date-array-header'],
            datetime_field: params['datetime-header'],
            enum_field: params['enum-header'],
        }
    }

    let echoUrlParams = async (params: service.EchoUrlParamsParams): Promise<models.UrlParameters> => {
        return {
            int_field: params.int_url,
            long_field: params.long_url,
            float_field: params.float_url,
            double_field: params.double_url,
            decimal_field: params.decimal_url,
            ool_field: params.bool_url,
            string_field: params.string_url,
            uuid_field: params.uuid_url,
            date_field: params.date_url,
            datetime_field: params.datetime_url,
            enum_field: params.enum_url,
        }
    }

    let echoEverything = async (params: service.EchoEverythingParams): Promise<service.EchoEverythingResponse> => {
        return {
            uuid_field: params['uuid-header'],
            datetime_field: params['datetime-header'],
            date_field: params.date_url,
            decimal_field: params.decimal_url,
            float_field: params.float_query,
            bool_field: params.bool_query,
            int_field: params.body.int_field, 
            string_field: params.body.string_field
        }
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }

    return {echoBodyString, echoBody, echoQuery, echoHeader, echoUrlParams, echoEverything, sameOperationName}
}